pyriemann_qiskit/utils/distance.py

```diff
@@ -4,26 +4,38 @@
 from pyriemann.classification import MDM
 from pyriemann.utils.distance import distance_functions
 from pyriemann.utils.base import logm
+from typing_extensions import deprecated
 
 
-def logeucl_dist_convex(X, y, optimizer=ClassicalOptimizer()):
-    """Convex formulation of the MDM algorithm with log-Euclidean metric.
+@deprecated(
+    "logeucl_dist_convex is deprecated and will be removed in 0.3.0; "
+    "please use distance_logeuclid_cpm."
+)
+def logeucl_dist_convex():
+    pass
+
+
+def distance_logeuclid_cpm(A, B, optimizer=ClassicalOptimizer()):
+    """Log-Euclidean distance to a convex hull of SPD matrices.
+
+    Log-Euclidean distance between a SPD matrix B and the convex hull of a set
+    of SPD matrices A [1]_, formulated as a Constraint Programming Model (CPM)
+    [2]_.
 
     Parameters
     ----------
-    X : ndarray, shape (n_classes, n_channels, n_channels)
+    A : ndarray, shape (n_matrices, n_channels, n_channels)
         Set of SPD matrices.
-    y : ndarray, shape (n_channels, n_channels)
-        A trial
+    B : ndarray, shape (n_channels, n_channels)
+        SPD matrix.
     optimizer: pyQiskitOptimizer
       An instance of pyQiskitOptimizer.
 
     Returns
```

Comment:
Since `distance_logeuclid_cpm` is a new function, you can modify its outputs:
it could return `weights` (for backward compatibility) but it must return the `distance` (because it is the name of the function).

Reply:
You mean not to modify the implementation, but just the description. Something like:

distances: ndarray, shape (n_matrices,)
        The distances between B and the set of SPD matrices A. 
        These distances are in fact the optimized weights for the set of SPD matrices A.
        Using these weights, one can compute a weighted Log-Euclidean mean of set A,
        that provides the matrix of the convex hull closest to matrix B.

Reply:
No, the idea is to add a new output.

Reply:
The other PR will be on implementing "QuantumNCH" in classyfication.py which will use `distance_logeuclid_cpm` which is the new distance to the hull. Is that correct?

Reply:
Yes, correct :)